model/task/task.go

```diff
@@ -151,6 +151,9 @@ type Task struct {
 	// Set to true if the task should be considered for mainline github checks
 	IsGithubCheck bool `bson:"is_github_check,omitempty" json:"is_github_check,omitempty"`
 
+	// CanReset indicates if the task is in a valid state to be reset.
+	CanReset bool `bson:"can_reset,omitempty" json:"can_reset,omitempty"`
```

Comment:
Would you mind adding a comment here or to the reset functions elaborating on this a bit? (i.e. that a valid state to be reset is that it's been archived successfully)